File: solutions/3101. Count Alternating Subarrays/3101.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def countAlternatingSubarrays(self, nums: List[int]) -> int:
    ans = 1
    size = 1

    for i in range(1, len(nums)):
      size = 1 if nums[i - 1] == nums[i] else size + 1
      ans += size

    return ans  

File: solutions/3101. Count Alternating Subarrays/3101.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  long long countAlternatingSubarrays(const vector<int>& nums) {
    long long ans = 1;
    int size = 1;
    
    for (int i = 1; i < nums.size(); ++i) {
      size = nums[i - 1] == nums[i] ? 1 : size + 1;
      ans += size;
    }
    
    return ans;
  }
};